src/app/clusters/group-key-mgmt-server/group-key-mgmt-cluster.cpp

```diff
@@ -102,208 +96,211 @@ struct GroupTableCodec
-        VerifyOrReturnError(nullptr != provider, CHIP_ERROR_INTERNAL);
+        return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
+    }
 
-        CHIP_ERROR err = aEncoder.EncodeList([provider](const auto & encoder) -> CHIP_ERROR {
-            CHIP_ERROR encodeStatus = CHIP_NO_ERROR;
+    return CHIP_NO_ERROR;
+}
 
-            for (auto & fabric : Server::GetInstance().GetFabricTable())
-            {
-                auto fabric_index = fabric.GetFabricIndex();
-                auto iter         = provider->IterateGroupInfo(fabric_index);
-                VerifyOrReturnError(nullptr != iter, CHIP_ERROR_NO_MEMORY);
+CHIP_ERROR ReadGroupTable(AttributeValueEncoder & aEncoder)
+{
+    auto provider = GetGroupDataProvider();
+    VerifyOrReturnError(nullptr != provider, CHIP_ERROR_INTERNAL);
 
-                GroupDataProvider::GroupInfo info;
-                while (iter->Next(info))
-                {
-                    encodeStatus = encoder.Encode(GroupTableCodec(provider, fabric_index, info));
-                    if (encodeStatus != CHIP_NO_ERROR)
-                    {
-                        break;
-                    }
-                }
-                iter->Release();
+    CHIP_ERROR err = aEncoder.EncodeList([provider](const auto & encoder) -> CHIP_ERROR {
+        CHIP_ERROR encodeStatus = CHIP_NO_ERROR;
+
+        for (auto & fabric : Server::GetInstance().GetFabricTable())
+        {
+            auto fabric_index = fabric.GetFabricIndex();
+            auto iter         = provider->IterateGroupInfo(fabric_index);
+            VerifyOrReturnError(nullptr != iter, CHIP_ERROR_NO_MEMORY);
+
+            GroupDataProvider::GroupInfo info;
+            while (iter->Next(info))
+            {
+                encodeStatus = encoder.Encode(GroupTableCodec(provider, fabric_index, info));
                 if (encodeStatus != CHIP_NO_ERROR)
                 {
                     break;
                 }
             }
-            return encodeStatus;
-        });
-        return err;
-    }
-    CHIP_ERROR ReadMaxGroupsPerFabric(EndpointId endpoint, AttributeValueEncoder & aEncoder)
+            iter->Release();
+            if (encodeStatus != CHIP_NO_ERROR)
+            {
+                break;
+            }
+        }
+        return encodeStatus;
+    });
+    return err;
+}
+
+CHIP_ERROR ReadMaxGroupsPerFabric(AttributeValueEncoder & aEncoder)
+{
+    auto * provider = GetGroupDataProvider();
+    VerifyOrReturnError(nullptr != provider, CHIP_ERROR_INTERNAL);
+    return aEncoder.Encode(provider->GetMaxGroupsPerFabric());
+}
+
+CHIP_ERROR ReadMaxGroupKeysPerFabric(AttributeValueEncoder & aEncoder)
+{
+    auto * provider = GetGroupDataProvider();
+    VerifyOrReturnError(nullptr != provider, CHIP_ERROR_INTERNAL);
+    return aEncoder.Encode(provider->GetMaxGroupKeysPerFabric());
+}
+
+bool GetProviderAndFabric(chip::app::CommandHandler * commandObj, const chip::app::ConcreteCommandPath & commandPath,
```

Comment:
```suggestion
bool GetProviderAndFabric(CommandHandler * commandObj, const ConcreteCommandPath & commandPath,
```

I know, pre-existing....  I do wonder why this got moved.